lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -345,21 +410,82 @@ def _discard_tunnel_flows(self, precedence:int, i_teid:int,
+                match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
+                                   in_port=self._uplink_port, **ip_match_out)
+
+                flows.delete_flow(self._datapath, self.tbl_num, match,
+                                  priority=priority +1)
+
+                match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
+                                   in_port=self.config.mtr_port, **ip_match_out)
+
+                flows.delete_flow(self._datapath, self.tbl_num, match,
+                                  priority=priority + 1)
+
+    def _discard_tunnel_ip_flow_dl(self, ip_flow_dl:IPFlowDL):
+        priority = Utils.get_of_priority(ip_flow_dl.precedence)
+        match = self._match_ip_flow_dl(ip_flow_dl, self._uplink_port)
+        flows.delete_flow(self._datapath, self.tbl_num, match, priority=priority+1)
+
+        match = self._match_ip_flow_dl(ip_flow_dl, self.config.mtr_port)
+        flows.delete_flow(self._datapath, self.tbl_num, match, priority=priority+1)
+
+    def _match_ip_flow_dl(self, ip_flow_dl:IPFlowDL, in_port:int):
+        ip_match_dst = get_ue_ip_match_args(ip_flow_dl.dest_ip, Direction.IN)
+        ip_match_src = get_ue_ip_match_args(ip_flow_dl.src_ip, Direction.OUT)
+
+        if ip_flow_dl.ip_proto == IPPROTO_TCP:
+            match = MagmaMatch(eth_type=get_eth_type(ip_flow_dl.dest_ip),
+                           in_port=in_port, **ip_match_dst,
+                           **ip_match_src, ip_proto=ip_flow_dl.ip_proto,
+                           tcp_src=ip_flow_dl.tcp_src_port,
+                           tcp_dst=ip_flow_dl.tcp_dst_port)
+
+        elif ip_flow_dl.ip_proto == IPPROTO_UDP:
+            match = MagmaMatch(eth_type=get_eth_type(ip_flow_dl.dest_ip),
+                               in_port=in_port, **ip_match_dst,
+                               **ip_match_src, ip_proto=ip_flow_dl.ip_proto,
+                               udp_src=ip_flow_dl.udp_src_port,
+                               udp_dst=ip_flow_dl.udp_dst_port)
+
+        return match
+
+    def gtp_handler(self, session_state, precedence:int, local_f_teid:int,
+                    o_teid:int, ue_ip_addr:IPAddress, gnb_ip_addr:str,
+                    sid:int = None, ue_ipv6_address:IPAddress = None,
+                    apn:str = None, vlan:int = 0, ip_flow_dl:IPFlowDL = None):
+
+        if (session_state == PdrState.Value('INSTALL')
+             or session_state == UESessionState.ACTIVE):
+            self.add_tunnel_flows(precedence, local_f_teid,
+                                  o_teid, ue_ip_addr,
+                                  gnb_ip_addr, sid, ue_ipv6_address,apn,
+                                  vlan, ip_flow_dl)
+
+        elif (session_state == PdrState.Value('IDLE')
+               or session_state == UESessionState.INSTALL_IDLE):
+            self.delete_tunnel_flows(local_f_teid, ue_ip_addr, gnb_ip_addr, ip_flow_dl)
+
+        elif (session_state == PdrState.Value('REMOVE')
```

Comment:
Looks like REMOVE/IDLE have the same remove operation, if that is intended combine them into one elif

Reply:
For IDLE case, delete tunnel operation and add paging operation will happen. For paging code, PR #3535 has been implemented.